libcxx/test/libcxx/containers/strings/basic.string/asan_deque_integration.pass.cpp

```diff
@@ -0,0 +1,183 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: asan
+// UNSUPPORTED: c++03
+
+#include <cassert>
+#include <array>
+#include "test_macros.h"
+#include "asan_testing.h" // includes deque and string - don't do it before
```

Comment:
What happens if we include deque or string before we include this header?

Reply:
I think nothing anymore, but I have to confirm. Testing code in deque used to be guarded by a macro, bu I don't see it anymore, this comment may be a leftover. I copied it from another ASan test.

If it works here, I'm going to remove this comment from here, but for other tests I will make a separate PR with also a small refactor of `libcxx/test/support/asan_testing.h`.
Probably after branching, unless you prefer us to do it faster.

Reply:
Yes, this comment is outdated and I'm removing it. It also actually does not appear in other tests. But `asan_testing.h` deserves a small refactor, which I'm planning to do soon.